File: 2021/08/19/add_digits.py
"""
Problem: https://leetcode.com/problems/add-digits/
Author: Andrew Jarombek
Date: 8/19/2021
"""


def add_digits(num: int) -> int:
    value = num

    while value >= 10:
        new_value = 0

        while value > 0:
            new_value += (value % 10)
            value //= 10

        value = new_value

    return value


if __name__ == '__main__':
    assert add_digits(38) == 2

File: 2023/02/04/go/add_digits.go
/**
 * https://leetcode.com/problems/add-digits/
 * Author: Andrew Jarombek
 * Date: 2/4/2023
 */

package _go

func addDigits(num int) int {
    result := num

    for result > 9 {
        n := result
        result = 0

        for n > 0 {
            result += n % 10
            n /= 10
        }
    }

    return result
}
